src/smt/proof_post_processor.cpp

```diff
@@ -83,7 +83,9 @@ bool ProofPostprocessCallback::shouldUpdate(std::shared_ptr<ProofNode> pn,
       && (id == ProofRule::TRUST_THEORY_REWRITE || id == ProofRule::TRUST))
   {
     d_trustedPfs.insert(pn);
-    return false;
+    // Note that we may fill in TRUST steps below. Thus, some proofs in
+    // d_trustedPfs may already be filled in when they are returned by this
+    // class.
```

Comment:
Can TRUST_THEORY_REWRITE be expanded? 

Also, to avoid storing things that you may expand, why not update d_trustedPfs during the expansion (trust step will always be attempted to be expanded)?

Reply:
And no, TRUST_THEORY_REWRITE is not expanded (caught by `shouldExpand` below).